pkg/controller/kabaneroplatform/targetnamespaces.go

```diff
@@ -0,0 +1,130 @@
+				APIGroup: "rbac.authorization.k8s.io",
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			Kind: "ClusterRole",
+			Name: "kabanero-pipeline-deploy-role",
+			APIGroup: "rbac.authorization.k8s.io",
+		},
+	}
+
+
+	// List of Namespaces we want to bind
+	targetnamespaceList := k.Spec.TargetNamespaces
+	
+	// If targetNamespaces is empty, default to binding to kabanero
+	if len(targetnamespaceList) == 0 {
+		targetnamespaceList = append(targetnamespaceList, "kabanero")
+	}
+
+	/* Get all Rolebindings named kabanero-pipeline-deploy-rolebinding
+		Structured method only lists RoleBindings in kabanero namespace. Maybe due to client scoping?
+		
+	rolebindingList := &rbacv1.RoleBindingList{}
+	err := cl.List(ctx, rolebindingList, client.MatchingFields{"metadata.name": "kabanero-pipeline-deploy-rolebinding"})
+	if err != nil {
+		return err
+	}
+	*/
+
+	// Get all Rolebindings named kabanero-pipeline-deploy-rolebinding
+	rolebindingList := &unstructured.UnstructuredList{}
+	rolebindingList.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "rbac.authorization.k8s.io",
+		Kind:    "RoleBindingList",
+		Version: "v1",
+	})
+	err := cl.List(ctx, rolebindingList, client.MatchingFields{"metadata.name": "kabanero-pipeline-deploy-rolebinding"})
+	if err != nil {
+		return err
+	}
+
+	// For each Rolebinding
+	for _, rolebinding := range rolebindingList.Items {
+		matchFound := false
+		// If the Rolebinding namespace is in the targetNamespace list
+		for i, targetnamespace := range targetnamespaceList {
+			if rolebinding.GetNamespace() == targetnamespace {
+			
+				// Fill in the namespace for the template
+				desiredRolebinding := rolebindingResource
+				desiredRolebinding.ObjectMeta.Namespace = targetnamespace
+			
+				// Check if the existing Rolebinding matches the desired template, and skip?
+				// Update may handle this already
+				
+				// Apply the rolebinding
+				cl.Update(ctx, &desiredRolebinding)
+				
+				// Remove the namespace from the list of Namespaces remaining to bind
+				targetnamespaceList[i] = targetnamespaceList[len(targetnamespaceList)-1] // Copy last element to index i.
+				targetnamespaceList[len(targetnamespaceList)-1] = ""                     // Erase last element (write zero value).
+				targetnamespaceList = targetnamespaceList[:len(targetnamespaceList)-1]   // Truncate slice.
+
+				matchFound = true
+				break
+			}
+		}
+
+		// If the Rolebinding does not match a targetNamespace, delete
+		if matchFound == false {
+			cl.Delete(ctx, &rolebinding)
+		}
+	}
+	
+	// For remaining namespaces in the list, Create the Rolebinding
+	for _, targetnamespace := range targetnamespaceList {
+		// Check if the namespace exists
+		namespace := &unstructured.Unstructured{}
+		namespace.SetGroupVersionKind(schema.GroupVersionKind{
+				Group:   "",
+				Kind:    "Namespace",
+				Version: "v1",
+		})
+		err = cl.Get(ctx, client.ObjectKey{Namespace: targetnamespace, Name: targetnamespace,}, namespace)
+		if err == nil {

```

Comment:
I guess I'm confused about this.... I would have thought that `cl.Get()` would return a NotFound error if it couldn't find it in the target namespace.  So you'd only want to create it if the NotFound was returned.

Reply:
This is a check that the namespace exists before attempting to create the RoleBinding within the namespace. The check isn't entirely necessary. Can just attempt to create the RoleBinding and let it fail, and I had it that way at one point. I'm not even sold that doing Get is the right way to check, but it doesn't seem like the client package has opinionated methods for such things.

Reply:
I'm sorry - I can't read :-) yes this is fine.